sdk/python/kfp/compiler/_component_builder.py

```diff
@@ -93,90 +93,56 @@ def generate_pip_requirements(self, target_file):
     """ write the python packages to a requirement file
     the generated file follows the order of which the packages are added """
     with open(target_file, 'w') as f:
-      for name, version in self.python_packages.items():
+      for name in sorted(self.python_packages.keys()):
+        version = self.python_packages[name]
         version_str = ''
         if version.has_min_version():
           version_str += ' >= ' + version.min_version + ','
         if version.has_max_version():
           version_str += ' <= ' + version.max_version + ','
         f.write(name + version_str.rstrip(',') + '\n')
 
-class DockerfileHelper(object):
-  """ Dockerfile Helper generates a tarball with dockerfile, ready for docker build
-      arc_dockerfile_name: dockerfile filename that is stored in the tarball """
-
-  def __init__(self, arc_dockerfile_name):
-    self._arc_dockerfile_name = arc_dockerfile_name
-    self._ARC_REQUIREMENT_FILE = 'requirements.txt'
-
-  def _generate_pip_requirement(self, dependency, requirement_filepath):
-    dependency_helper = DependencyHelper()
-    for version in dependency:
-      dependency_helper.add_python_package(version)
-    dependency_helper.generate_pip_requirements(requirement_filepath)
+def _dependency_to_requirements(dependency=[], filename='requirements.txt'):
+  """
+    Generates a requirement file based on the dependency
+    Args:
+      dependency (list): a list of VersionedDependency, which includes the package name and versions
+      filename (str): requirement file name, default as requirements.txt
+  """
+  dependency_helper = DependencyHelper()
+  for version in dependency:
+    dependency_helper.add_python_package(version)
+  dependency_helper.generate_pip_requirements(filename)
 
-  def _generate_dockerfile_with_py(self, target_file, base_image, python_filepath, has_requirement_file, python_version):
-    """ _generate_docker_file generates a simple dockerfile with the python path
-    args:
-      target_file (str): target file name for the dockerfile.
+def _generate_dockerfile(filename, base_image, entrypoint_filename, python_version, requirement_filename=None):
```

Comment:
We probably do not need to set the entrypoint at all since it can/should be set by the component's `command`.

Reply:
This is the entrypoint filename that gets copied to the image. We can discuss offline and address it in the next PR.